### PR TITLE
Pairwise aligner save more memory

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request implements memory-efficient versions of the Needleman-Wunsch, Smith-Waterman, Gotoh global and local, and Waterman-Smith-Beyer global and local algorithm. Alignable sequences are approximately 5 times longer.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/Align/__init__.py [hunk 1]
```diff
@@ -28,6 +28,7 @@
     new_exc.__cause__ = None
     raise new_exc
 
+from Bio.Align import _aligners_old
```
Comment (peterjc) [inline Bio/Align/__init__.py]: I don't see this code - is this as the Python classes using it temporary debugging code?

**Comment by peterjc** [comment]
This looks like a big change worth adding a NEWS.rst entry about the support for Gotoh algorithm.

Also it looks like ``Bio/Align/_aligners.c`` has conflicts, most likely from #1844. I personally would use a git rebase to sort that out. Right now that's blocking the automated tests from taking place.

**Comment by mdehoon** [comment]
Starting anew

**Comment by mdehoon** [comment]
@peterjc  Perhaps it looks impressive from the diff, but actually this is just some reorganization of the memory management code in this module to allow the algorithms to run with much less memory. But the algorithms themselves have not changed, and the Gotoh algorithm was already in there.

**Comment by peterjc** [comment]
The refactoring made it look from a quick skin the the Gotoh work was new - my mistake- thanks

**Pull request closed without merging**